twine/utils.py

```diff
@@ -28,6 +28,9 @@
 from urllib.parse import urlunparse
 
 import requests
+from rfc3986 import exceptions as rfc_exceptions
+from rfc3986 import uri_reference
+from rfc3986 import validators
```

Comment:
I think importing the whole module is more aligned with the [twine's code style](https://twine.readthedocs.io/en/latest/contributing.html#code-style):

```diff
--- a/twine/utils.py
+++ b/twine/utils.py
@@ -30,5 +30,3 @@ from urllib.parse import urlunparse
 import requests
-from rfc3986 import exceptions as rfc_exceptions
-from rfc3986 import uri_reference
-from rfc3986 import validators
+import rfc3986
 
@@ -110,3 +108,3 @@ def validate_url(repository_url: str) -> None:
     validator = (
-        validators.Validator()
+        rfc3986.validators.Validator()
         .allow_schemes("http", "https")
@@ -115,3 +113,3 @@ def validate_url(repository_url: str) -> None:
 
-    url = uri_reference(repository_url)
+    url = rfc3986.uri_reference(repository_url)
     validator.validate(url)
@@ -128,3 +126,3 @@ def get_repository_from_config(
             validate_url(repository_url)
-        except rfc_exceptions.MissingComponentError as exc:
+        except rfc3986.exceptions.MissingComponentError as exc:
             raise exceptions.UnreachableRepositoryURLDetected(
@@ -132,3 +130,3 @@ def get_repository_from_config(
             )
-        except rfc_exceptions.UnpermittedComponentError as exc:
+        except rfc3986.exceptions.UnpermittedComponentError as exc:
             raise exceptions.UnreachableRepositoryURLDetected(
```

This passes the tests locally.

Reply:
Thanks. I have changed the import in the latest commit